src/tests/catwalk_tests/test_baselines.py

```diff
@@ -83,6 +84,71 @@ def test_predict_proba(self):
             np.testing.assert_array_equal(results, expected_results)
 
 
+@pytest.mark.usefixtures("data")
+class TestRankMultiFeature(TestCase):
+    def test_fit(self):
+        rules = {'feature': 'x3', 'low_value_high_score': False}
+        ranker = BaselineRankMultiFeature(rules=rules)
+        assert ranker.feature_importances_ is None
+        ranker.fit(x=self.data["X_train"], y=self.data["y_train"])
+        np.testing.assert_array_equal(
+            ranker.feature_importances_, np.array([0, 0, 1, 0])
+        )
+
+    def test_ranking_on_unavailable_feature_raises_error(self):
+        rules = [{'feature': 'x5', 'low_value_high_score': False}]
+        ranker = BaselineRankMultiFeature(rules=rules)
+        with self.assertRaises(BaselineFeatureNotInMatrix):
+            ranker.fit(x=self.data["X_train"], y=self.data["y_train"])
+
+    def test_predict_proba_one_feature(self):
+        for direction_value in [True, False]:
+            rules = {'feature': 'x3', 'low_value_high_score': direction_value}
+            ranker = BaselineRankMultiFeature(rules=rules)
+            ranker.fit(x=self.data["X_train"], y=self.data["y_train"])
+            results = ranker.predict_proba(self.data["X_test"])
+            if direction_value:
+                expected_ranks = [6, 1, 3, 0, 5, 2, 4, 5]
```

Comment:
I wonder if a clearer way to express this test (and one less prone to rewriting if the test fixture has to be modified for any reason) would be to order by the rank and assert that each one has a lower or equal/higher or equal value than the one prior.

I have mixed thoughts on this approach of duplicating logic in the test from the code under test; after all, if it's been implemented incorrectly in the code under test, who's to say the test doesn't have the same bad logic and thus erroneously pass? But I'm guessing in this case you wouldn't be able to copy and paste code from the code under test, and it probably wouldn't have that problem.

I'm not sure this is a big issue here. In general it's good to go for low maintenance tests that aren't beholden to what the test fixtures specifically look like, especially when the test fixtures are shared between different tests. You could in the future end up adding a new baseline class that needs more entropy than this fixture provides, and the data changes you make could require all these tests to be updated. What is the probability of needing to make such a change in the future? I don't know.  But in general this type of thing happens a lot, and it makes the tests seem like a burden so it's good to minimize when we can.

As far as the data goes, I might be misunderstanding the logic used; shouldn't the calls with inverse direction values but the same data have inverse expected ranks?

Reply:
Thanks, @thcrock! That's a good point -- I just added a helper function to check that the returned scores align to the expected ordering without having to reproduce the logic from the underlying function itself. Let me know if that looks better to you or if you think it could be cleaned up a bit more?

I think the calls with the different direction values do get the inverse expected ranks, but let me know if I'm missing something. For instance:
```
  if direction_value:
      expected_ranks = [6, 1, 3, 0, 5, 2, 4, 5]
  else:
     expected_ranks = [0, 5, 3, 6, 1, 4, 2, 1]
```